internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/NettyP2PBlobTransferManager.java

```diff
@@ -46,16 +49,19 @@ public CompletionStage<InputStream> get(String storeName, int version, int parti
     if (discoverPeers == null || discoverPeers.isEmpty()) {
       throw new VenicePeersNotFoundException("No peers found for the requested blob");
     }
-    try {
-      // TODO: add some retry logic or strategy to choose the peers differently in case of failure
-      // instanceName comes as a format of <hostName>_<applicationPort>
-      String chosenHost = discoverPeers.get(0).split("_")[0];
-      inputStream = nettyClient.get(chosenHost, storeName, version, partition);
-    } catch (Exception e) {
-      throw new VenicePeersNotFoundException("The connection to peers failed", e);
+    for (String peer: discoverPeers) {
+      try {
+        // TODO: add some retry logic or strategy to choose the peers differently in case of failure
+        // instanceName comes as a format of <hostName>_<applicationPort>
+        String chosenHost = peer.split("_")[0];
+        inputStream = nettyClient.get(chosenHost, storeName, version, partition);
+        LOGGER.info("Chosen host: {}", chosenHost);
```

Comment:
how about moving it above line 57? When there's something wrong happen at line 57, the line 58 would not get executed. though eventually It would also get logged at line 61